src/lib/components/Swap/Settings/MaxSlippageSelect.tsx

```diff
@@ -104,41 +111,28 @@ export default function MaxSlippageSelect() {
   const [autoSlippage, setAutoSlippage] = useAtom(autoSlippageAtom)
   const [maxSlippage, setMaxSlippage] = useAtom(maxSlippageAtom)
   const maxSlippageInput = useMemo(() => maxSlippage?.toString() || '', [maxSlippage])
-  const [warning, setWarning] = useState(WarningState.NONE)
+  const [warning, setWarning] = useState<WarningState | undefined>()
   const [showTooltip, setShowTooltip, tooltipProps] = useTooltip()
 
-  const processInput = useCallback(
-    (input: number | undefined) => {
-      const numerator = input && Math.floor(input * 100)
-      if (numerator) {
-        const percent = new Percent(numerator, 10_000)
-        if (percent.greaterThan(MAX_VALID_SLIPPAGE)) {
-          setWarning(WarningState.INVALID_SLIPPAGE)
-          setAutoSlippage(true)
-          setMaxSlippage(input)
-        } else if (percent.greaterThan(MIN_HIGH_SLIPPAGE)) {
-          setWarning(WarningState.HIGH_SLIPPAGE)
-          setAutoSlippage(false)
-          setMaxSlippage(input)
-        } else {
-          setWarning(WarningState.NONE)
-          setAutoSlippage(false)
-          setMaxSlippage(input)
-        }
-      } else {
-        setAutoSlippage(true)
-        setMaxSlippage(undefined)
+  const processValue = useCallback(
+    (value: number | undefined, isInput: boolean) => {
+      const percent = toPercent(value)
+      const warning = toWarningState(percent)
+      setWarning(warning)
+      if (isInput) {
+        setMaxSlippage(value)
+        setAutoSlippage(!percent || warning === WarningState.INVALID_SLIPPAGE)
       }
     },
     [setAutoSlippage, setMaxSlippage]
   )
   const onInputSelect = useCallback(() => {
     focus()
-    processInput(maxSlippage)
-  }, [focus, maxSlippage, processInput])
+    processValue(maxSlippage, /*isInput=*/ true)
```

Comment:
I don't immediately follow what this comment is trying to tell me
can you move it above and explain what's going on here?